From 0d2e56b70f9d8567b8cc86006163f7e2a1c9aff2 Mon Sep 17 00:00:00 2001
From: Xianzhong <b07117@freescale.com>
Date: Wed, 23 Jan 2013 20:09:14 +0800
Subject: [PATCH 409/635] ENGR00241376 Fixed opengl angles test crash when
 exit

The crash when exiting is because of recursive call in gralloc wrapper

Signed-off-by: Xianzhong <b07117@freescale.com>
---
 mx6/libgralloc_wrapper/gralloc.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mx6/libgralloc_wrapper/gralloc.cpp b/mx6/libgralloc_wrapper/gralloc.cpp
index 2ad03a2..bb96a42 100755
--- a/mx6/libgralloc_wrapper/gralloc.cpp
+++ b/mx6/libgralloc_wrapper/gralloc.cpp
@@ -1,6 +1,6 @@
 /*
  * Copyright (C) 2008 The Android Open Source Project
- * Copyright (C) 2010-2012 Freescale Semiconductor, Inc.
+ * Copyright (C) 2010-2013 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -157,7 +157,7 @@ static int gralloc_close(struct hw_device_t *dev)
          */
         private_module_t* m = reinterpret_cast<private_module_t*>(ctx->device.common.module);
         if(m && m->gpu_device){
-           gralloc_close((struct hw_device_t *)m->gpu_device);
+           m->gpu_device->common.close((struct hw_device_t *)m->gpu_device);
            m->gpu_device = NULL;
         }
 
-- 
1.8.0

